src/python/pants/backend/javascript/subsystems/nodejs.py

```diff
@@ -284,8 +353,21 @@ async def node_process_environment(
             search_path=nodejs.executable_search_path,
         ),
     )
+    corepack_env_vars = await Get(EnvironmentVars, EnvironmentVarsRequest(nodejs.corepack_env_vars))
+    binary_digest_with_shims = await add_corepack_shims_to_digest(
+        binaries, binary_shims, corepack_env_vars
+    )
```

Comment:
Corepack shims has to be part of the same digest as the downloaded binaries, otherwise the symlinks are incredibly incorrect when passed as separate immutable digests.